ui/src/containers/VolumePage.js

```diff
@@ -0,0 +1,750 @@
+    (vIOPSW) =>
+      vIOPSW.metric.instance === instance &&
+      vIOPSW.metric.device === volume?.status?.deviceName,
+  );
+  const volumeUsed = volumeUsedList?.filter(
+    (vU) => vU.metric.persistentvolumeclaim === PVCName,
+  );
+
+  return currentVolumeName && volume ? (
+    <PageContainer>
+      {/* There are two cases could be redirected to the Volume Page. */}
+      <BreadcrumbContainer>
+        {node.name !== undefined ? (
+          <Breadcrumb
+            activeColor={theme.brand.secondary}
+            paths={[
+              <BreadcrumbLabel title={intl.translate('platform')}>
+                {intl.translate('platform')}
+              </BreadcrumbLabel>,
+              <StyledLink to="/nodes">{intl.translate('nodes')}</StyledLink>,
+              <StyledLink to={`/nodes/${node.name}`} title={node.name}>
+                {node.name}
+              </StyledLink>,
+              <BreadcrumbLabel title={intl.translate('volumes')}>
+                {intl.translate('volumes')}
+              </BreadcrumbLabel>,
+            ]}
+          />
+        ) : (
+          <Breadcrumb
+            activeColor={theme.brand.secondary}
+            paths={[
+              <BreadcrumbLabel title={intl.translate('platform')}>
+                {intl.translate('platform')}
+              </BreadcrumbLabel>,
+              <BreadcrumbLabel title={intl.translate('volumes')}>
+                {intl.translate('volumes')}
+              </BreadcrumbLabel>,
+            ]}
+          />
+        )}
+      </BreadcrumbContainer>
+      <VolumeContent>
+        <LeftSideVolumeList>
+          <VolumeListContainer>
+            <Table columns={columns} data={volumeListData} rowInfo={rowInfo} />
+          </VolumeListContainer>
+        </LeftSideVolumeList>
+        <RightSidePanel>
+          <VolumeDetailCard
+            name={currentVolumeName}
+            nodeName={volume?.spec?.nodeName}
+            storage={pV?.spec?.capacity?.storage ?? intl.translate('unknown')}
+            status={volumeStatus ?? intl.translate('unknown')}
+            storageClassName={volume?.spec?.storageClassName}
+            creationTimestamp={volume?.metadata?.creationTimestamp}
+            volumeType={
+              volume?.spec?.rawBlockDevice
+                ? RAW_BLOCK_DEVICE
+                : SPARSE_LOOP_DEVICE
+            }
+            usedPodName={UsedPod?.name}
+            devicePath={
+              volume?.spec?.rawBlockDevice?.devicePath ??
+              intl.translate('not_applicable')
+            }
+            volumeUsagePercentage={currentVolume?.usage}
+            volumeUsageBytes={currentVolume?.usageRawData ?? 0}
+            storageCapacity={
+              volumeListData?.find((vol) => vol.name === currentVolumeName)
+                .storageCapacity
+            }
+            health={
+              volumeListData?.find((vol) => vol.name === currentVolumeName)
+                .health
+            }
```

Comment:
Same comment than for `PerformanceGraphCard ` *but I'm less sure*.
It may be a good thing to leave it that way. Can you have a look ?

Reply:
IMO, I think we could leave it in this way since we just pass some data to the detail card component. This is no big difference if we fetch the state in detail card component or volume page component.

Reply:
When not sure whether to implement a component as pure display or as smart / stateful, I try to think about lifecycle. More specifically, about comparing the lifecycle of the component vs. the lifecycle of the data it uses.

Let's take an example: say I'm displaying the Node Details tab component. This component's data lives with the Node object, and for the most part (except the metrics), will not change a lot. The component itself lives much shorter: switching between tabs will recreate new components everytime (which can be even shorter than metrics refresh rate).
In this case, I would build the tab component to be purely about display (dumb component), and make sure the parent would provide it with a fresh Node object (including its stats).

(as you may have guessed, this means that IMO, neither of the components aforementioned should be smart, outside the "data extraction" logic they embed)

Related to [the discussion above](https://github.com/scality/metalk8s/pull/2665#discussion_r462082994), I think the reasoning was biased. The symptom @MonPote observed is well summed up in:

> It's a 1 to 1 relationship and add a lots of noise to the component for nothing

This is entirely true, and we should think of a solution to this problem.

But the proposed one is only solving it by side-effect: if the `PerformanceGraphCard` is a container, it means we change the ownership of data, and as such, will not need to worry about passing down values anymore since the component will fetch it on its own once mounted.
Another solution we could consider to the above problem, without changing ownership of data, is to change the contract between parent and child: pass a full `volumeStats` object as a prop to `PerformanceCard`, and let it handle the extraction of information to display (specific time series for each graph it needs based on known keys), and let the parent (`VolumePage`) do the mapping between a `volume` object and the relevant `volumeStats` required by the performance card.

A "dumb component" is not as dumb as to need flattened information. "Dumb" is to distinguish components managing state (performing creative/destructive actions on application state, from user operations or external data consumption) versus components consuming this state managed for them. It doesn't mean you can't write code outside of HTML/JSX in them ;)

The only component in this page which makes sense to be a container is the `VolumePage`. It manages a list of volumes, and dispatches relevant information to each display components it shows, maintaining a cached (though probably refreshed over time) view of each volume.

One could argue that we don't want to refresh stats for an unselected volume, especially if we have a lot of them (though honestly not sure that would make a big impact if we group everything in a single query to Prometheus). If that's the case, then the next level would be to consider a component taking care of "one selected volume" (which we don't have in the current implementation). This component would be made a container, able to fetch and maintain the metrics for a volume which was passed by its parent, the `VolumePage`.

However, let's consider the lifecycle question again: the `VolumePage` component receives the URL `match` as a prop. If another volume than the current one is selected, the URL will change, and the component will be re-rendered (which, in this case, also means dispatching some actions to the store which in turn will send calls to the K8s API). A big chunk of `VolumePage` managed data (the list of all volumes, the selected filters, the active tab), on the other hand, would (probably) outlive the change between selected Volumes.
So... we need to make this page a "dumb" component as well? Who takes care of retrieving those volumes in that case? IMO, that's a sign that a container is owning a piece of state that it should not (in this case, the selected Volume name).

Imagine if the `VolumePage` was mounted and kept intact for all URLs (`useRouteMatch("/volumes")`). This container can manage a list of volumes, handle modals and their interaction with sagas (and I would argue for it to also maintain stats, but this can be debated, so let's say it won't for now, it will make the example more interesting). This container also renders a sub-router component, which in turn will match more specifically on the rest of the route (`useRouteMatch("/volumes/:id")`) and handle volume-specific information (such as metrics), which don't need to be maintained when the volume isn't selected.
Now you can work with a single source of the list of volumes, managed by the page container. A child container manages the selected volume (through URL) and receives the list of volumes from its parent. This child container can mount a dumb component for the table on the left, highlighting the selected volume, and another child container for the right-side panel, responsible for refreshing live information about the selected volume it received as a prop.

I hope this long message wasn't too long and that it made my point clear. Let's sum it up:
- For the performance card component, making it a container would have created a lifecycle mismatch (managed data lives longer than the component), and we solved it by bubbling up the state management in its parent component.
- For the Volume page component, we had the opposite mismatch (the information of which volume is selected lives, or should live, shorter than the full Volume page), but couldn't solve it by the same method. Instead, we reduced the scope of managed data in the parent and bubbled down the state management into a child container.